### ISSUE TITLE
Would Error Messages be helpful for C API extensions ?

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#454]
### SketchUp C API Issue

**Would Error Messages be helpful (at runtime) for C API extensions ?**

A file like the following ...

```c
// "error_message.h"

#ifndef ERROR_MESSAGE_H_
#define ERROR_MESSAGE_H_

static const char *ErrorMessage[]= {
  "No error / Successful",                // SU_ERROR_NONE
  "Required input pointer was null",      // SU_ERROR_NULL_POINTER_INPUT, 
  "Invalid object input to function",     // SU_ERROR_INVALID_INPUT, 
  "Ouput pointer null / count is null",   // SU_ERROR_NULL_POINTER_OUTPUT
  "Invalid object for function output",   // SU_ERROR_INVALID_OUTPUT
  "Attempt to overwrite valid reference", // SU_ERROR_OVERWRITE_VALID
  "Generic / Unspecified error",          // SU_ERROR_GENERIC
  "File serialization IO error",          // SU_ERROR_SERIALIZATION
  "Out of range error",                   // SU_ERROR_OUT_OF_RANGE
  "No data to return",                    // SU_ERROR_NO_DATA
  "Insufficient size to receive output",  // SU_ERROR_INSUFFICIENT_SIZE
  "Unknown exception",                    // SU_ERROR_UNKNOWN_EXCEPTION
  "Invalid model",                        // SU_ERROR_MODEL_INVALID
  "Model version invalid",                // SU_ERROR_MODEL_VERSION
  "Layer/Tag locked",                     // SU_ERROR_LAYER_LOCKED
  "Duplicate data operation",             // SU_ERROR_DUPLICATE
  "Partial success",                      // SU_ERROR_PARTIAL_SUCCESS
  "Unsupported operation",                // SU_ERROR_UNSUPPORTED
  "Invalid argument",                     // SU_ERROR_INVALID_ARGUMENT
  "Entity locked",                        // SU_ERROR_ENTITY_LOCKED
  "Invalid operation"                     // SU_ERROR_INVALID_OPERATION
};

#endif  // ERROR_MESSAGE_H_
```
... could be used by an error handler function. It could look up a message using the return code from most of the C API functions as an index into the constant array of messages.

Thoughts ?

**Comment by thomthom** [comment]
The messages seems to spell out what the enum value says. Not seeing a whole not of difference here - just duplicated information to maintain.

Description for each enum value is documented:
https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/common_8h.html#a2f14198fe6741c51570c7c666c32a674

Any decent IDE will display this info when you hover over it.

Or am I missing a use-case here?

**Comment by DanRathbun** [comment]
> The messages seems to spell out what the enum value says.

Well, I didn't want to deviate too much from the original intent (as I'm not that "comfy" with the internals of the C API.) **Sure, perhaps they could use improvement.**

> Or am I missing a use-case here?

In some code I was playing with I had a function that used the string returned to output to stdout.
It could also be used for messageboxes. (Ie, the returned string could be a substring of some information that is displayed.)

Basically was thinking of something similar to how Ruby's exception inspection string gives a nice statement containing the source file, the line, the method, the exception type and the error message.

> Description for each enum value is documented: ...
> Any decent IDE will display this info when you hover over it.

I did see this. I'm not sure what "hover over it" means. Realize I'm coming at this from a **runtime** point of view, not from a browsing the API header files in an IDE point of view.

Updated the question in the first post to ...
"**Would Error Messages be helpful (at runtime) for C API extensions ?**"
